src/bucket/BucketManager.h

```diff
@@ -207,9 +207,11 @@ class BucketManager : NonMovableOrCopyable
     virtual void maybeSetIndex(std::shared_ptr<Bucket> b,
                                std::unique_ptr<BucketIndex const>&& index) = 0;
 
-    virtual std::unique_ptr<SearchableBucketListSnapshot const>
+    virtual std::unique_ptr<SearchableBucketListSnapshot>
```

Comment:
While I agree in principal, the only members of `SearchableBucketListSnapshot` are `mSnapshotManager`, which is already const, and `std::unique_ptr<BucketListSnapshot const> mSnapshot{};`. Because `SearchableBucketListSnapshot` checks and automatically updates `mSnapshot` on each load, to be const correct `mSnapshot` would have to be mutable. I don't think this makes much sense to have the only piece of non-const state be mutable, and I don't think it's dangerous because the underlying `BucketListSnapshot` is const regardless.